src/trace-api.ts

```diff
@@ -241,6 +242,17 @@ export class StackdriverTracer implements Tracer {
             rootSpan.span.name}] was already closed.`);
         return UNCORRELATED_CHILD_SPAN;
       }
+      if (rootSpan.trace.spans.length === this.config!.spansPerTraceSoftLimit) {
+        // As in the previous case, a root span with a large number of child
+        // spans suggests a memory leak stemming from context confusion. This
+        // is likely due to userspace task queues or Promise implementations.
+        this.logger!.warn(`TraceApi#createChildSpan: [${
```

Comment:
The problem I see with this is that warnings are not shown to the user by default because the default log level is 'error'.

I would propose making this an error (if you do this alongside my other suggestion of increasing the value by an order of magnitude).

Reply:
I agree, if the limit is high enough we should just emit an error. However, I'm not sure if we should change behavior by setting a hard upper limit to # spans per trace.

Reply:
I changed the PR to add a hard limit. PTAL